storage/innobase/row/row0ins.cc

```diff
@@ -1279,9 +1279,8 @@ row_ins_foreign_check_on_constraint(
 						 clust_index, tmp_heap);
 	}
 
-	if (cascade->is_delete && foreign->v_cols != NULL
-	    && foreign->v_cols->size() > 0
-	    && table->vc_templ == NULL) {
+	if (cascade->is_delete && dict_table_get_n_v_cols(table) > 0
+			&& table->vc_templ == NULL) {
 		innobase_init_vc_templ(table);
 	}
```

Comment:
how about fixing the row_ins_foreign_fill_virtual()? I think it is not being called to set out the template initialization?

Reply:
yes. It is not called for same reason . we have a check  that[ foreign key](https://github.com/percona/percona-server/blob/a73ccf393d8c41410b7f25f657cc178b2d860a05/storage/innobase/row/row0ins.cc#L1330) should have the virtual column.

Reply:
can we remove this part of the code and fix that instead? ie call row_ins_foreign_fill_virtual() based on child virtual columns presence?

Reply:
we do not call row_ins_foreign_fill_virtual for this code path. It is called in case DICT_FOREIGN_ON_DELETE_SET_NULL. here the code flow DICT_FOREIGN_ON_DELETE_CASCADE

Reply:
ok then row_ins_foreign_fill_virtual() becomes dummy operation? Can you please check with SET_NULL and see if row_ins_foreign_fill_virtual() is called again? And this double initialization is safe?

Reply:
@rahulmalik87 did you check?

Reply:
SET_NULL doens't go inside this method since we have additional check [cascade->delete](https://github.com/percona/percona-server/blob/a73ccf393d8c41410b7f25f657cc178b2d860a05/storage/innobase/row/row0ins.cc#L1282)

Reply:
Rahul, am asking you check with constraint ON DELETE  SET NULL

Reply:
in case of ON DELETE SET NULL. cascade->delete is set to 0. 
lldb) p cascade->is_delete
(ulint) $1 = 0
(lldb) p node->is_delete
(ulint) $2 = 1
(lldb)

Reply:
cascade delete as 1 with ON DELETE as NULL not possible?

Reply:
ON DELETE either SET NULL or Cascade. so CASCADE and SET NUL is not possible